multiregion: Added a section for multiregion parameterized job dispatch (#14664)
hashicorp/nomad: website/content/docs/job-specification/multiregion.mdx

state where it waits until the last region has completed the deployment. The
final region will unblock the regions to mark them as `successful`.

## Multiregion Parameterized Job Dispatch

Job dispatching is region specific. While a [parameterized job] can be 

> deployed to multiple [federated regions] like any other job, a parameterized

Reviewer: ```suggestion
registered in multiple [federated regions] like any other job, a parameterized
```

As we've learned, deploy/deployment is a little overused in Nomad 😅

job operates much like a function definition that takes variable input.
Operators are expected to invoke the job by invoking [`job dispatch`] 
from the CLI or the [HTTP API] and provide the appropriate dispatch options
for that region.

> 

Reviewer: It may be worth adding a note about periodic jobs here as well. They will run at the same "global" time right? Like 5PM UTC, or whatever timezone you set in https://developer.hashicorp.com/nomad/docs/job-specification/periodic#time_zone

[examples]: #multiregion-examples
[upgrade strategies]: https://learn.hashicorp.com/collections/nomad/job-updates
[`nomad deployment unblock`]: /docs/commands/deployment/unblock
[parameterized job]: /docs/job-specification/parameterized
[`job dispatch`]: /docs/commands/job/dispatch

> [HTTP API]: /api-docs/jobs

Reviewer: ```suggestion
[HTTP API]: /api-docs/jobs#dispatch-job
```
